src/types/resolveDescriptors.ts

```diff
@@ -1006,6 +1027,95 @@ export function resolveDescriptors(ctx: CompilerContext) {
                             },
                             ast: d,
                         });
+                    } else if (
+                        d.selector.kind === "internal-const-comment" ||
+                        d.selector.kind === "external-const-comment"
+                    ) {
+                        const internal =
+                            d.selector.kind === "internal-const-comment";
+
+                        let path = d.selector.comment;
```

Comment:
It feels like the following constant look-up code should be factored out into a helper function. This also should be implemented for expressions as well, can't we reuse that code here?

Reply:
Btw, the constant lookup you have implemented here, should go to the constant evaluator: https://github.com/tact-lang/tact/blob/9ed9193e81dc3ed240a1f6d2530659d2705b7ba2/src/constEval.ts#L521. So, after typechecking the receiver constant string, you just run const-eval on it and you should get your string. This way you will also extend the supported language fragment for const-eval and resolve issue #229.

Reply:
We could even allow arbitrary expressions in receivers as long as those are string-valued constant expressions

Reply:
The problem is that this way we'll try to evaluate a constant that might depend on the resolved contract during resolving that contract. With static constants it'll work fine, but with cases like `receive(s.something)` it won't work...

Reply:
Why not? We can first process all constants and then process receivers

Reply:
Because these are stored in CompilerContext with the contract itself in the very end of the resolving:
https://github.com/tact-lang/tact/blob/9ed9193e81dc3ed240a1f6d2530659d2705b7ba2/src/types/resolveDescriptors.ts#L1622-L1624

So I can't get contract-scoped constants by calling `getAllTypes` or `getType` in constant evaluator.

Reply:
if we cannot move it before receiver processing stage, let's just push this PR to v1.5 and solve this properly